### ISSUE TITLE
`Style/SingleArgumentDig`: false positive when my own class has a method with the name dig.

### ISSUE THREAD

**Issue by KarlHeitmann** [rubocop/rubocop#11534]
I have my own class, and I named its own method `#dig`

It appears Rubocop has detected the call to `#dig` with a single argument and
complains because it is better to use single brackets instead of calling dig with only
one argument.

I could turn off that rule, but I am also using `Hash` class and `#dig` method in other parts
of the code.

Is it possible to fix this issue? So that Rubocop can detect if the object is a `Hash` or not, and complain
only if the object is a `Hash`?

--------

## Expected behavior

It does not detect offense of `Style/SingleArgumentDig`.

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

```
$ rubocop --debug --cache false --only Style/SingleArgumentDig offended.rb

For /home/karl/github: configuration from /home/karl/github/.rubocop.yml
Default configuration from /home/karl/github/rubocop/config/default.yml
Inspecting 1 file
Scanning /home/karl/github/offended.rb
C

Offenses:

offended.rb:12:6: C: [Correctable] Style/SingleArgumentDig: Use f[:a] instead of f.dig(:a).
puts f.dig(:a)
     ^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.08871467800054234 seconds

```

## Steps to reproduce the problem

```rubocop.yml
# rubocop.yml
AllCops:
  TargetRubyVersion: 2.7
  NewCops: disable

```

```ruby
# offended.rb
class Foo
  def initialize
    @calls = []
  end

  def dig(*args)
    @calls << args
    rand
  end
end

puts f.dig(:a)
puts f.dig(:a, :b)

```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop/exe/rubocop --version
1.44.1
```

**Comment by fatkodima** [comment]
This cop is already marked as unsafe - https://github.com/rubocop/rubocop/blob/5230ef2238a7c88c96ee512251d248f987d49f74/lib/rubocop/cop/style/single_argument_dig.rb#L9-L12 so this is expected.

> Is it possible to fix this issue? So that Rubocop can detect if the object is a Hash or not, and complain
only if the object is a Hash?

Rubocop does static analysis of the code and, unfortunately, is unable to know which type some variable will have at runtime.
